core/src/test/scala/unit/kafka/log/UnifiedLogTest.scala

```diff
@@ -2580,6 +2580,23 @@ class UnifiedLogTest {
     assertEquals(None, log.leaderEpochCache.flatMap(_.latestEpoch.toScala))
   }
 
+  @Test
+  def testLeaderEpochCacheCreatedAfterMessageFormatUpgrade(): Unit = {
```

Comment:
This test was originally removed via #18267 - added it back and modified it to take into account that the leader epoch cache always exists now (since the configured record version is always V2).